engine/src/main/java/com/sandboni/core/engine/sta/connector/TestSuiteConnector.java

```diff
@@ -0,0 +1,36 @@
+package com.sandboni.core.engine.sta.connector;
+
+import com.sandboni.core.engine.sta.Context;
+import com.sandboni.core.engine.sta.graph.Link;
+import com.sandboni.core.engine.sta.graph.LinkFactory;
+import com.sandboni.core.engine.sta.graph.LinkType;
+import com.sandboni.core.engine.sta.graph.vertex.TestSuiteVertex;
+import com.sandboni.core.engine.sta.graph.vertex.Vertex;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Connects between each suite test class to it's related methods (test vertices)
+ */
+public class TestSuiteConnector implements Connector {
+
+    @Override
+    public void connect(Context context) {
+        Set<Vertex> tests = context.getLinks().filter(l -> l.getLinkType() == LinkType.ENTRY_POINT).map(Link::getCallee).collect(Collectors.toSet());
+        Set<Vertex> allTestSuiteVertices = context.getLinks().filter(l -> l.getLinkType() == LinkType.TEST_SUITE).map(Link::getCallee).collect(Collectors.toSet());
+        // connect test suite to relevant test (if applicable):
+        tests.forEach(t -> {
+            // check if we have a related suite class
+            allTestSuiteVertices.parallelStream().filter(tsv -> tsv instanceof TestSuiteVertex && ((TestSuiteVertex)tsv).getRelatedTestClasses().contains(t.getActor())).forEach(ts -> {
```

Comment:
In a standard project, you could have just a few allTestSuiteVertices but a lot of tests. In this case you are splitting suites to match with tests, but if you have only 1 or 2 suites, your gain will be minimum if you split to execute in parallel. I would rather split the tests (tests.parallelStream().forEach) and let the allTestSuiteVertices in a simple stream.